### DISCUSSION TITLE
Ability to block/ban/blacklist a device from joining the network

### DISCUSSION THREAD

**Discussion by jtbandes** [zigpy/zigpy#1320] [Ideas]
A couple of my neighbor’s Hue lights are happily joining my ZHA network whenever I enable joining by going to the “add device” page. If I remove them, they will simply rejoin the next time I try to add a new device to my network. I’ve read that in Z2M [there is a way to ban specific devices](https://community.home-assistant.io/t/deleted-zigbee-device-wont-show-up-again/629660/2) from joining the network.

It would be great if this would also be possible in zigpy / ZHA. (I realize that I could allow the devices to join but keep them disabled — but it seems preferable to not allow them to join in the first place.)

I have filed a ZHA feature request on the Home Assistant forum here as well: https://community.home-assistant.io/t/ability-to-block-blacklist-ban-a-device-from-joining-zha-by-ieee-address/668816

**Comment by puddly** [comment]
There's unfortunately no way to _prevent_ joins but it's possible to provide a way to ignore the devices and politely ask them to leave if they try.

**Reply by jtbandes** [reply]
Yes, this seems to be what zigbee2mqtt does:

https://www.zigbee2mqtt.io/guide/configuration/block-pass-list.html

https://github.com/Koenkk/zigbee2mqtt/blob/9eaaa0f9d1725416639ee894fc45542be3a4753f/lib/zigbee.ts#L313-L315

https://github.com/Koenkk/zigbee-herdsman/blob/faed33ef36fde00df48513dd2eab9b33222f539d/src/controller/controller.ts#L532-L539

**Comment by 1liminal1** [comment]
Did this ever go anywhere?

**Comment by Hakun1n** [comment]
This is a must. The amount of "smart" devices in peoples houses is growing every day. In most of the cases they're using BT and controlling the stuff via cellphone. So they're not even aware about Zigbee. I also have bunch of foreign bulbs re-appearing in my network. If i delete them, they will come back in few hours. All of this because i went into pairing mode once to connect new device and this cr4p automatically joined my network...

**Reply by MattWestb** [reply]
If the "ghost" device is good behaving its possible doing one automation with sending one leave command (without rejoin) and i have getting one IKEA CWS 600 going then its working as standard. But one HUE is not possible then the is not deleting the network credentials so can rejoinga all the time so have making one automation that is putting it to 1% brightness so not working for the owner (normally its coming online 2-4 in the night).
Then ZHA is not possible locking down the network with install code (that 95% of all normal devices not supporting) its the only way distorting the network and forming one new and adding all 100 ++ devices.
Good for my is that its looks the HUE and its owner have moving away or putting the device there its belong (in the Junk yard) so its being under control for my and i moving all sensors that is possible to Thread and have the ZHA production  system primary for lights that need Zigbee binding IKEA Parasoll
for controlling the light then open the door to the under storage is guarantee working 100% also is the network is down.

**Reply by Hakun1n** [reply]
Ah, thanks, i will give it a try. The "ghost" devices are some "Eglo ZM RGB TW by AwoX" lights. Forming new network with new key is annoying solution (migrate dozen of devices). Also it only takes one situation when i hit "Add Device" during the time the foreign devices are in pairing mode and they're back again.
I will try the "leave" command next time they get back and lets see.
Messing around with the foreign devices sounds "fun" but i will leave it as a last thing to try ...
